### PR TITLE
Sinatra Tutorial.

### PR DESCRIPTION
**Pull request by lpil**
Because I totally broke the last PR.

ping @deniseyu

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -40,62 +40,67 @@
...
-            } catch(err) {}
-</script>
+        <ul>
+          <li><a href="sinatra/lesson1/tutorial.html">Lesson 1 - Introduction to Sinatra</a></li>
+        </ul>
+
+        <h2>Python</h2>
+
+        <ul>
+          <li><a href="python/lesson1/tutorial.html">Lesson 1 - Introduction to Python</a></li>
```
Comment (despo) [inline index.html]: the link to the second Python tutorial is missing ` <li><a href="python/lesson2/tutorial.html">Lesson 2 - Playing with variables</a></li>`

**Comment by despo** [comment]
@lpil looks good!

File: sinatra/lesson1/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+## Installing Ruby
+
+Before you begin, you must have Ruby installed on your machine. There are a
+couple of different ways you can get ruby setup.
+
+- Windows installation using [Ruby installer][ruby-installer]
+- Mac & Linux installation using [rbenv][rbenv-install]
+
+[rbenv-install]: https://github.com/sstephenson/rbenv "rbenv"
+[ruby-installer]: http://rubyinstaller.org/
```
Comment (jonnyarnold) [inline sinatra/lesson1/tutorial.md]: If you've done the Ruby tutorial (line 12/13), you'll have Ruby installed.

File: sinatra/lesson1/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+typically call our libraries "gems", purely because it's a cute name.
+
+To save us the trouble of copy and pasting library code into our projects,
+applications were written that will go and download the libraries you ask for,
+and then make them available in your project. These applications are called
+"package managers", and our Ruby package manager is called "gem".
+
+So when you ran `gem install sinatra` before you were instructing the `gem`
+package manager to go and download the `sinatra` gem so you could use it in
+your project without any extra fuss. How convenient!
```
Comment (jonnyarnold) [inline sinatra/lesson1/tutorial.md]: Great explanation. Is there something like this in the Ruby tutorial? If not, we should move it there.

File: sinatra/lesson1/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+```
+Stopping ...
+== Sinatra has ended his set (crowd applauds)
+```
+
+Refresh the page in your web browser- it's no longer available. Web servers,
+like any other computer program, only do what we tell them to do, and only work
+while they are running.
+
+## Routes and URLs
```
Comment (jonnyarnold) [inline sinatra/lesson1/tutorial.md]: Perhaps explain routes:

> `get '/' do` in the example above is an instruction for Sinatra that can be read as 'if the user tries to **get** `http://localhost:4567/` (with an ending **/**), then **do** the following'. This is known as a *route*, which can simply be thought of as a bit of Ruby code that runs when someone makes a request to your server.

I'm not sure if that's better, now I've written it. I just think the term route hasn't been defined well.

File: sinatra/lesson1/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+and this new one runs when we `get` a request to the `/about` address.
+
+Start up our server again with `ruby app.rb`, and navigate to
+`http://localhost:4567/about`- we've created a new page. Navigate back to
+`http://localhost:4567/`, our first page is still there.
+
+### Mini challenge!
+
+Try making a few more routes for URLs of your choice, for example
+`http://localhost:4567/kittens`
```
Comment (jonnyarnold) [inline sinatra/lesson1/tutorial.md]: Another challenge:

> To send a file, you can end your route with `send_file 'foo.png'`, as long as `foo.png` exists in your web server's folder. Add a route that sends you a picture when you access it with your web browser!

File: sinatra/lesson1/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+## End of part one
+
+So today we've created our own simple web application using Ruby and Sinatra,
+and learnt a little bit about servers, routing, and the web. If there's
+anything you'd like explained in more detail, pop down to codebar, and ask one
+of the excellent coaches.
```
Comment (jonnyarnold) [inline sinatra/lesson1/tutorial.md]: I'd add a Further Reading section:

> The whole documentation for Sinatra is on a single (rather long) web page. [Go have a look](http://www.sinatrarb.com/intro.html): you've covered the *Routes* section in this chapter, and you can see that there's plenty more to come!

**Comment by jonnyarnold** [comment]
I <3 Sinatra. It should be bigger than Rails in the 'just learning' community, because there's less magic.

**Pull request closed without merging**